src/compiler/checker.ts

```diff
@@ -10494,17 +10512,24 @@ namespace ts {
         }
 
         function getContextualTypeForJsxAttribute(attribute: JsxAttribute | JsxSpreadAttribute) {
+            // When we trying to resolve JsxOpeningLikeElement as a stateless function element, we will already give JSXAttributes a contextual type
+            // which is a type of the parameter  of the signature we are trying out. This is not the case if it is a statefull Jsx (i.e ReactComponenet class)
+            // So if that is the case, just return the type of the JsxAttribute in such contextual type with out going into resolving of the JsxOpeningLikeElement again
+            if ((<JsxAttributes>attribute.parent).contextualType) {

```

Comment:
Actually, as we looked at it together, I think you can simplify it even more with something like
```ts
const attributesType = getContextualType(attribute.parent) || getAttributesTypeFromJsxOpeningLikeElement(attribute.parent.parent);
if (kind === SyntaxKind.JsxAttribute) ...
```